src/zcbor_encode.c

```diff
@@ -141,26 +141,35 @@ static bool value_encode(zcbor_state_t *state, zcbor_major_type_t major_type,
 }
 
 
-bool zcbor_int32_put(zcbor_state_t *state, int32_t input)
-{
-	return zcbor_int64_put(state, input);
-}
-
-
-bool zcbor_int64_put(zcbor_state_t *state, int64_t input)
+bool zcbor_int_encode(zcbor_state_t *state, const void *input_int, size_t int_size)
 {
 	zcbor_major_type_t major_type;
+	uint8_t input_buf[8];
+	const uint8_t *input_uint8 = input_int;
+	const int8_t *input_int8 = input_int;
+	const uint8_t *input = input_int;
+
+	if (int_size > sizeof(int64_t)) {
```

Comment:
So far, all instances of `int_size` come from `sizeof()` calls, so they are strictly speaking known at compile time. I don't know how I would set it up to do the compile time check, though, since this code should work for different sizes.